specs/Print.md

```diff
@@ -0,0 +1,816 @@
+  DisplayPdfDataInPrintDialog(stream);
+
+  MessageBox.Show(this, "Printing " + title + " document to PDF Stream succeeded", "Print To PDF Stream");
+}
+
+// Function to display current web page pdf data in a custom print preview dialog.
+void DisplayPdfDataInPrintDialog(Stream pdfData)
+{
+  // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
+  /// Opens the browser print preview dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_BROWSER,
+
+  /// Opens the system print dialog.
+  COREWEBVIEW2_PRINT_DIALOG_KIND_SYSTEM,
+
+} COREWEBVIEW2_PRINT_DIALOG_KIND;
+
+/// Specifies the duplex option for a print.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DUPLEX {
+  /// The default duplex for a printer.
+  COREWEBVIEW2_PRINT_DUPLEX_DEFAULT,
+
+  /// Print on only one side of the sheet.
+  COREWEBVIEW2_PRINT_DUPLEX_ONE_SIDED,
+
+  /// Print on both sides of the sheet, flipped along the long edge.
+  COREWEBVIEW2_PRINT_DUPLEX_TWO_SIDED_LONG_EDGE,
+
+  /// Print on both sides of the sheet, flipped along the short edge.
+  COREWEBVIEW2_PRINT_DUPLEX_TWO_SIDED_SHORT_EDGE,
+
+} COREWEBVIEW2_PRINT_DUPLEX;
+
+/// Specifies the color mode for a print.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_COLOR_MODE {
+  /// The default color mode for a printer.
+  COREWEBVIEW2_PRINT_COLOR_MODE_DEFAULT,
+
+  /// Indicate that the printed output will be in color.
+  COREWEBVIEW2_PRINT_COLOR_MODE_COLOR,
+
+  /// Indicate that the printed output will be in shades of gray.
+  COREWEBVIEW2_PRINT_COLOR_MODE_GRAYSCALE,
+
+} COREWEBVIEW2_PRINT_COLOR_MODE;
+
+/// Specifies the collation for a print.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_COLLATION {
+  /// The default collation for a printer.
+  COREWEBVIEW2_PRINT_COLLATION_DEFAULT,
+
+  /// Indicate that the collation has been selected for the printed output.
+  COREWEBVIEW2_PRINT_COLLATION_COLLATED,
+
+  /// Indicate that the collation has not been selected for the printed output.
+  COREWEBVIEW2_PRINT_COLLATION_UNCOLLATED,
+
+} COREWEBVIEW2_PRINT_COLLATION;
+
+/// Specifies the media size for a print.
+[v1_enum] typedef enum COREWEBVIEW2_PRINT_MEDIA_SIZE {
+  /// The default media size for a printer.
+  COREWEBVIEW2_PRINT_MEDIA_SIZE_DEFAULT,
+
+  /// Indicate custom media size that is specific to the printer.
+  COREWEBVIEW2_PRINT_MEDIA_SIZE_CUSTOM,
+
+} COREWEBVIEW2_PRINT_MEDIA_SIZE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : ICoreWebView2_14 {
+  /// Print the current web page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings.
+  /// The handler will return `errorCode` as `S_OK` and `isSuccessful` as FALSE if `printerName`
```

Comment:
Really subtle communication. Better to make `isSuccessful` an enum rather than a bool, and have enum values for all of the expected errors rather than relying on the HR. (Using the HR means that in the C# projection we have expected errors that an app is supposed to catch.)

Reply:
PrintAsync - errorCode & isSuccessful (bool)

PrintAsync, change isSuccessful into CoreWebView2PrintStatus enum with 'Succeeded' (match WebErrorStatus's name) (hr == S_OK, isSuccessful == True), 'PrinterUnavailable' (hr == S_OK, isSuccessful == False), 'OtherError' (error == E_FAIL). When enum is used, errorCode is always S_OK. 
E_INVALIDARG & E_ABORT remain as exceptions (via errorCode - and enum is OtherError)

PrintToPdfStreamAsync - errorCode & stream pointer
Doesn't have the same case because those other failure modes only apply with a real pinter & spooler and such.

ShowPrintUI - no errors (no synchronous errors) - This seems OK as distinct from the other two considering the different nature of this method how its showing printing UI. Need to document what happens when the dialog is already up or a different kind of dialog is already up.